students/mkdir01/lesson04/mailroom2.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+import sys
+
+# changed from list of lists to a dict with values as list
+data = {'John Doe': [120.00, 353.33, 400.00], 'Jane Doe': [3500.04, 2624.44], 'John Smith': [33.33, 400.00, 29.20], 'Jane Smith': [2.00], 'Billy Jo Jones': [100.00, 100.00, 100.00]}
+
+
+# mode 1 - "send a thank you"
+def thank_you():
+    while True:
+        name = input("Please enter a full name (or type \'list\', or hit \'return\' to go back)> ")
+        if name == "list":
+            for k in data:
+                print(k)
+        elif name == "":
+            return
+        else:
+            dollars = input("Please enter a donation amount (ex: 500.05)> ")
+            for k in data:
+                if k == name:
+                    data[k].append(round(float(dollars), 2))
+                    break
+            else:  # if no name match, make new entry
+                data[name] = [(round(float(dollars), 2))]
+            print(f"Dear {name},\n\nThank you for your donation of {data[name][-1]:.2f}.\n\nLove,\nThe Charity Org\n")
+            return
+
+
+# mode 2 - "create a report"
+def sort_by_dollars(donor_lists):  # bubble sort
+    counter = 0  # counter will ensure we've taken one full pass of the list where all rows are in order before quitting
+    temp_row = []
+    while counter < (len(donor_lists) - 1):  # we will get an out of bounds error if we don't stop at len-1
+        counter = 0
+        for i in range(len(donor_lists) - 1):
+            if donor_lists[i][1] < donor_lists[i + 1][1]:
+                temp_row = donor_lists.pop(i)
+                donor_lists.insert(i + 1, temp_row)
+            else:
+                counter += 1  # when we get through a whole for loop withought "bubbling" we leave the while loop
+    else:
+        return donor_lists
```

Comment:
I would still recommend using `sorted` instead of custom sort functions. 

it is usually preferred to use something already in place instead of reimplementing for these reasons:
* readability: reader is most likely familiar with the `sorted` function and easy to follow
* avoids bugs: a built-in function or a common library most likely not to have bugs due to its wide usage.

Reply:
let me know if you need a practical example how to fit `sorted` into your code.